### ISSUE TITLE
[Typescript-Angular] Generated code for OpenAPI3.0 spec can not be build/packaged

### ISSUE THREAD

**Issue by ity42** [OpenAPITools/openapi-generator#1185]
##### Description

Multiple messages like: "Cannot find module '../model/uNKNOWNBASETYPE'." while doing npm run build in client-api-folder, resulting in an Exit Status 111.

##### openapi-generator version

https://oss.sonatype.org/content/repositories/snapshots/org/openapitools/openapi-generator-cli/3.3.1-SNAPSHOT/openapi-generator-cli-3.3.1-20181005.160335-9.jar

##### OpenAPI declaration file content or url

https://gist.githubusercontent.com/amilos/1ce55dbdfa336eee1de74d3e800496c1/raw/2400bc8072378afcfb707c72f908883bbfa0e077/bg-psd2.yaml

##### Command line used for generation

$wget https://oss.sonatype.org/content/repositories/snapshots/org/openapitools/openapi-generator-cli/3.3.1-SNAPSHOT/openapi-generator-cli-3.3.1-20181005.160335-9.jar

$java -jar openapi-generator-cli-3.3.1-20181005.160335-9.jar generate \\
   -i https://gist.githubusercontent.com/amilos/1ce55dbdfa336eee1de74d3e800496c1/raw/2400bc8072378afcfb707c72f908883bbfa0e077/bg-psd2.yaml \\
   -g typescript-angular \\
   -o ./psd2-client-api \\
   -c openapi-generator-config.json

(with openapi-generator-config.json: 
{
  "npmName": "@ity/psd2-api",
  "npmVersion": "0.0.1",
  "snapshot": true,
  "ngVersion": "6.0.0"
}
)

$cd psd2-client-api

$npm install

$npm run build
=> resulting in multiple times "Cannot find module '../model/uNKNOWNBASETYPE'."

##### Steps to reproduce

see "Command line used for generation"

##### Related issues/PRs

Have not found a related issue

##### Suggest a fix/enhancement

Sorry, these are my first steps with OpenAPI and the openapi-generator. I do not have any further insights.

**Comment by julianazuniga** [comment]
Do you have the package.json file? I can not even run npm install

**Comment by macjohnny** [comment]
is this still an issue with `4.0.0`?

**Comment by svatwork** [comment]
Any workaround?

**Comment by paquettea** [comment]
Ran into issue by trying to import the generated-sources directly from the file system in my Angular project.
Instead I had to package theses sources and refer to them as a dependency in node_module:
- ran 'npm pack' in the generated sources 
- referred the generated tgz file as a dependency of my Angular project (package.json), 
- added the name of the package as "allowedNonPeerDependencies" (ng-package.json)
- ran npm install in my angular project
- added the import for ApiModule in my AppModule
- built the app 

It's probably due to some tsconfig that are different and required to be loaded properly, without packing the app, it's the angular configuration that tries to build the files in generated source, instead of the ts config created by the generated command. I do not understand all of it as of now, but packaging the generated sources then installing them does it for me. Hope this helps someone
